File: android/src/main/kotlin/com/example/audio_graph/IdManager.kt
package com.example.audio_graph

class IdManager(val key: String) {
    companion object {
        private val ids: MutableMap<String, Int> = mutableMapOf()

        fun generateId(key: String): Int {
            val id = ids[key] ?: 0
            ids[key] = id + 1
            return id
        }
    }

    fun generateId(): Int {
        return IdManager.generateId(key)
    }
}

File: ios/Classes/IdManager.swift
//
//  IdManager.swift
//  audio_graph
//
//  Created by Kaisei Sunaga on 2020/01/24.
//

import Foundation

class IdManager {
  private static var ids: [String: Int] = [:]
  
  static func generateId(for key: String) -> Int {
    if let value = ids[key] {
      ids[key] = value + 1
      return value + 1
    } else {
      ids[key] = 0
      return 0
    }
  }
  
  private let key: String
  
  init(key: String) {
    self.key = key
  }
  
  func generateId() -> Int {
    return IdManager.generateId(for: key)
  }
}
